src/legacy/core_plugins/kibana/public/dashboard/grid/dashboard_grid.tsx

```diff
@@ -182,18 +181,18 @@ class DashboardGridUi extends React.Component<Props, State> {
           : PanelUtils.parseVersion('6.0.0');
 
       if (panelVersion.major < 6 || (panelVersion.major === 6 && panelVersion.minor < 1)) {
-        panel = PanelUtils.convertPanelDataPre_6_1(panel as Pre61PanelState);
+        panel = PanelUtils.convertPanelDataPre_6_1((panel as unknown) as Pre61SavedDashboardPanel);
```

Comment:
the reason for this casting is that we are replacing these panels in place, which we really shouldn't be doing.  So we go from an array of panels with multiple shapes and at the end have an array with panels only of the most recent shape.  Doing a type every where of `SavedDashboardPanel | Pre61SavedDashboardPanel` is more trouble than this is worth.  Ideally we should introduce a saved object migration to get rid of all this manual migration jazz that was added before saved object migrations were possible.

Reply:
> Ideally we should introduce a saved object migration to get rid of all this manual migration jazz that was added before saved object migrations were possible.

(note to myself) - Could probably be addressed in #35864